bundler/spec/support/helpers.rb

```diff
@@ -330,7 +330,7 @@ def with_built_bundler(version = nil)
 
         replace_version_file(version, dir: build_path) # rubocop:disable Style/HashSyntax
 
-        gem_command! "build bundler.gemspec", :dir => build_path
+        gem_command! "build #{shipped_gemspec}", :dir => build_path
```

Comment:
This doesn't make sense to me. We must be in the same folder as the gemspec in order to build the gem correctly, and we have just copied the file in there. We want the gemspec to be to copied one (where we have changed the version), not the one from the sources.

Reply:
I understood. But It's hard to rewite the filelist on only path.rb. Can I add the `ruby_core?` condition in helpers.rb?

Reply:
I don't understand what you mean. The rewritten file list comes from `shipped_files` in `path.rb`  so it should be correct for the core repo, shouldn't it?

Reply:
gemspec in ruby repo was located in `lib/bundler/bundler.gemspec`, not `bundler.gemspec` in `shipped_files`. so, build_path + "/bundler.gemspec" was failed with `bundle build bundler.gemspec`.

We should move it to the top of `build_path`

Reply:
Like this https://github.com/ruby/ruby/pull/3105/files

Reply:
I understand now. The patch you linked to makes sense :+1:.